google-cloud-firestore/CONTRIBUTING.md

```diff
@@ -1,187 +1,390 @@
 # Contributing to Google Cloud Firestore
 
-1. **Sign one of the contributor license agreements below.**
-2. Fork the repo, develop and test your code changes.
-3. Send a pull request.
+Thank you for your interest in making a contribution to google-cloud-ruby. Community contributions are an essential part
+of open source, and we want to make contributing easy for you. If you have any suggestions for how to improve this
+guide, please [open an issue](https://github.com/googleapis/google-cloud-ruby/issues) and let us know!
 
-## Contributor License Agreements
+### Code of Conduct
 
-Before we can accept your pull requests you'll need to sign a Contributor
-License Agreement (CLA):
+Please note that this project is covered by a Contributor Code of Conduct. By participating in this project you agree to
+abide by its terms. See [Code of Conduct](CODE_OF_CONDUCT.md) for more information.
 
-- **If you are an individual writing original source code** and **you own the
-  intellectual property**, then you'll need to sign an [individual
-  CLA](https://developers.google.com/open-source/cla/individual).
-- **If you work for a company that wants to allow you to contribute your work**,
-  then you'll need to sign a [corporate
+## Overview
+
+1. [Open an issue](#open-an-issue)
+1. [Sign Contributor License Agreement](#sign-contributor-license-agreement)
+1. [Set up environment](#set-up-environment)
+1. [Run CI](#run-ci)
+1. [Make changes](#make-changes)
+1. [Commit changes](#commit-changes)
+1. [Run CI again](#run-ci-again)
+1. [Submit your pull request](#submit-your-pull-request)
+
+## Open an issue
+
+Pull requests should generally be directed by an existing issue, otherwise you risk working on something that the
+maintainers might not be able to accept into the project. Please take a look through [the repository
+issues](https://github.com/googleapis/google-cloud-ruby/issues), and if you do not see an existing issue for your
+problem or feature, please open one using one of the provided templates.
+
+## Sign Contributor License Agreement
+
+Before we can accept your pull requests you'll need to sign a Contributor License Agreement (CLA):
+
+- **If you are an individual writing original source code** and **you own the intellectual property**, then you'll need
+  to sign an [individual CLA](https://developers.google.com/open-source/cla/individual).
+- **If you work for a company that wants to allow you to contribute your work**, then you'll need to sign a [corporate
   CLA](https://developers.google.com/open-source/cla/corporate).
 
-You can sign these electronically (just scroll to the bottom). After that, we'll
-be able to accept your pull requests.
+You can sign these electronically. After that, we'll be able to accept your pull requests.
 
-## Setup
+## Set up environment
 
-In order to use the google-cloud-firestore console and run the project's tests,
-there is a small amount of setup:
+To set up your local development environment:
```

Comment:
Maybe consider sub-headings (`###`) for these three (development environment, acceptance tests, interactive console)

Reply:
Will do.

Reply:
This turned out to be a great suggestion as it allowed me to make the internal links more specific.